project-base/src/Shopsys/ShopBundle/Resources/scripts/frontend/components/ajaxMoreLoader.js

```diff
@@ -1,29 +1,32 @@
 (function ($) {
 
     Shopsys = window.Shopsys || {};
-    Shopsys.productList = Shopsys.productList || {};
-    Shopsys.productList.AjaxMoreLoader = Shopsys.productList.AjaxMoreLoader || {};
+    Shopsys.AjaxMoreLoader = Shopsys.AjaxMoreLoader || {};
 
-    Shopsys.productList.AjaxMoreLoader = function () {
+    Shopsys.AjaxMoreLoader = function ($wrapper) {
         var self = this;
         var $loadMoreButton;
-        var $currentProductList;
+        var $currentList;
         var $paginationToItemSpan;
 
         var totalCount;
         var pageSize;
         var page;
+        var pageQueryParameter;
         var paginationToItem;
+        var url;
 
         this.init = function () {
-            $loadMoreButton = $('.js-load-more-button');
-            $currentProductList = $('.js-product-list');
-            $paginationToItemSpan = $('.js-pagination-to-item');
+            $loadMoreButton = $wrapper.filterAllNodes('.js-load-more-button');
+            $currentList = $wrapper.filterAllNodes('.js-list');
+            $paginationToItemSpan = $wrapper.filterAllNodes('.js-pagination-to-item');
 
             totalCount = $loadMoreButton.data('total-count');
             pageSize = $loadMoreButton.data('page-size');
             page = $loadMoreButton.data('page');
+            pageQueryParameter = $loadMoreButton.data('page-query-parameter');
```

Comment:
Maybe make this also optional?
```suggestion
            pageQueryParameter = $loadMoreButton.data('page-query-parameter') || 'page';
```